src/platform/backends/shared/linux/backend_utils.cpp

```diff
@@ -103,15 +104,31 @@ void check_min_img_size(const mp::MemorySize& requested_size, const mp::Path& im
     const auto pattern = QStringLiteral("^virtual size: .+ \\((?<size>\\d+) bytes\\)$");
     const auto re = QRegularExpression{pattern, QRegularExpression::MultilineOption};
 
-    if (const auto match = re.match(img_info); match.hasMatch())
+    const auto match = re.match(img_info);
+
+    if (!match.hasMatch())
+        throw std::runtime_error{fmt::format("Could not obtain image's virtual size")};
+
+    const auto min_size = mp::MemorySize(match.captured("size").toStdString());
+
+    // If the size is zero at this point, it means that the user did not specify a size.
+    // We need to resize only if the default size is bigger than the image size.
+    if (0 == requested_size.in_bytes())
```

Comment:
It would be cleaner if it was an `mp::optional`, I suppose?

Reply:
Yeah, `mp::optional` will solve our problems.